### ISSUE TITLE
Avg loss of new model is not less than the avg loss of the old model even when the old model has not seen new data points.

### ISSUE THREAD

**Issue by musram** [VowpalWabbit/vowpal_wabbit#3981]
Hi,

The initial model is trained as
/tmp/vw --data /tmp/data.dat -f /tmp/cb.model --hash all --bit_precision 28 -k --passes 5 --onethread -q CH -q CA -q HA -q RA --cb_explore_adf --cb_type dr --save_resume --holdout_off --cache_file .

Then the new model is trained as
 /tmp/vw --data /tmp/data.dat -i /tmp/cb.model  -f /tmp/cb_new.model --hash all --bit_precision 28 -k --passes 5 --onethread -q CH -q CA -q HA -q RA --cb_explore_adf --cb_type dr --save_resume --holdout_off --power_t 0 --cache_file.

Only if the avg loss of the new model is less than the initial model on the test data, do we update the old model with the new model.( https://vowpalwabbit.org/docs/vowpal_wabbit/python/latest/tutorials/off_policy_evaluation.html)

--power_t 0 is used so that the learning rate is not decreased while retraining the old model.

We are finding that the old model is not getting replaced by the new model which should not happen as the model is seeing some of the data which are new to the model. 

I checked the learning rate which is 0.5. The output of training new model is as below
learning rate = 0.5
initial_t = 0
power_t = 0
decay_learning_rate = 1

I feel I am missing something here? Can anybody help? 

@JohnLangford @jackgerrits

**Comment by musram** [comment]
@JohnLangford 

For evaluation, I am splitting the production data (generated using the old model policy) into the train (80%) and test (20%).
old_model_loss_on_test = avg cost of the test.

I retrain the old model with the train data and then evaluate it using the test data. The avg loss while evaluating the new model on the test is used to compare with the old_model_loss_on_test.

As I am using test data separately I pass  --holdout_off in the command line.

**Comment by JohnLangford** [comment]
The other obvious possibility is that your data source is nonstationary.   Is that plausible?  In particular, if you permute your training events and repeat the experiments, what happens?

**Comment by musram** [comment]
@JohnLangford 
I shuffled the train data and repeated the experiments. But the avg loss remains the same.

Nonstationary is plausible as the relation between the input features and the target label may change over time. To address this issue,  we have taken --power_t 0 as mentioned in https://vowpalwabbit.org/docs/vowpal_wabbit/python/latest/tutorials/cmd_linear_regression.html.

**Comment by JohnLangford** [comment]
Is the new model's perf worse than the old model's on the test set when both old and new are trained on permuted data?  

If so, it's interesting.  If not, it's just saying that non-stationarity is plausibly the culprit.

**Comment by jackgerrits** [comment]
@musram Is this issue resolved by identifying the non-stationarity?

**Comment by musram** [comment]
@jackgerrits it's resolved.
